nucliadb_node/src/shards/providers/shard_cache/reader_cache.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2021 Bosutech XXI S.L.
+//
+// nucliadb is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at info@nuclia.com.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex, MutexGuard};
+
+use nucliadb_core::{node_error, NodeResult};
+
+use super::resource_cache::{CacheResult, ResourceCache};
+use crate::disk_structure;
+use crate::settings::Settings;
+use crate::shards::errors::ShardNotFoundError;
+use crate::shards::metadata::ShardMetadata;
+use crate::shards::reader::ShardReader;
+use crate::shards::ShardId;
+
+pub struct ShardReaderCache {
```

Comment:
This could just be a couple of functions. I decided to keep it as an struct for symmetry with the ShardWriter, which works better with an struct to keep track of shards under deletion.